core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -211,6 +211,7 @@ void handleUncaughtThrowable(Throwable t) {
   @CheckForNull
   private final ChannelTracer channelTracer;
   private final Channelz channelz;
+  private Boolean nonzeroBackends; // a flag for doing channel tracing when flipped
```

Comment:
I think this part would be more clear if we created a separate `enum` for resolved and error, since the goal is to avoid consecutive reports for the same event type.

Reply:
Per the new spec, error events in general are not logged unless it results in # of backends down to zero from nonzeor. (Both "zero backends resolved" and "general error" are considered # of backends <- zero). A separate `enum` for error seems overly specific. Maybe the flag can be refactored as "error"?